test/fhirtypes/ElementDefinition.constrainType.test.ts

```diff
@@ -338,6 +339,46 @@ describe('ElementDefinition', () => {
       );
     });
 
+    it('should allow us to constrain a reference to a profile with a parent with a canonical including the version', () => {
+      const hasMember = observation.elements.find(e => e.id === 'Observation.hasMember');
+      const hasMemberConstraint = new OnlyRule('hasMember');
+      hasMemberConstraint.types = [
+        {
+          type: 'https://fhir.kbv.de/StructureDefinition/KBV_PR_Base_Observation_Body_Weight',
+          isReference: true
+        }
+      ];
+      hasMember.constrainType(hasMemberConstraint, fisher);
+      expect(hasMember.type).toHaveLength(1);
+      expect(hasMember.type[0]).toEqual(
+        new ElementDefinitionType('Reference').withTargetProfiles(
+          'https://fhir.kbv.de/StructureDefinition/KBV_PR_Base_Observation_Body_Weight'
+        )
+      );
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
+    });
+
+    it('should log an error when referencing a profile with a parent with a wrong version', () => {
```

Comment:
And maybe this is a little better?
```suggestion
    it('should log an error when constraining a reference to a profile whose parent is specified using the wrong version', () => {
```